vaxrank/vaccine_peptide.py

```diff
@@ -15,60 +13,58 @@
 
 from __future__ import absolute_import, print_function, division
 
-from collections import namedtuple
 from operator import attrgetter
 
 import numpy as np
+from serializable import Serializable
 
 from .manufacturability import ManufacturabilityScores
 
-VaccinePeptideBase = namedtuple(
-    "VaccinePeptide", [
-        "mutant_protein_fragment",
-        "mutant_epitope_predictions",
-        "wildtype_epitope_predictions",
-        "mutant_epitope_score",
-        "wildtype_epitope_score",
-        "num_mutant_epitopes_to_keep",
-        "manufacturability_scores"])
-
 
-class VaccinePeptide(VaccinePeptideBase):
+class VaccinePeptide(Serializable):
     """
     VaccinePeptide combines the sequence information of MutantProteinFragment
     with MHC binding predictions for subsequences of the protein fragment.
 
-    The resulting lists of mutant and wildtype epitope predictions are sorted by ic50.
+    The resulting lists of mutant and wildtype epitope predictions
+    are sorted by affinity.
     """
-    def __new__(
-            cls,
+
+    def __init__(
+            self,
             mutant_protein_fragment,
             epitope_predictions,
-            num_mutant_epitopes_to_keep=10000):
+            num_mutant_epitopes_to_keep=10000,
+            sort_predictions_by='ic50'):
```

Comment:
what would you want to sort by other than ic50?

Reply:
The only other current option would `percentile_rank`

Reply:
My hope is to expose everything that's hard-coded as parameters here and eventually have them all in a config file, instead of buried in code.

Reply:
(including the currently default args to logistic_epitope_score)